src/main/java/vazkii/quark/content/management/module/AutomaticToolRestockModule.java

```diff
@@ -224,12 +225,20 @@ private void switchItems(Player player, QueuedRestock restock) {
 		ItemStack stackAtPlayerSlot = playerInv.getItem(playerSlot).copy();
 		ItemStack stackProvidingSlot = providingInv.getStackInSlot(providingSlot).copy();
 
+		//Botania rods are only detected in the stackAtPlayerSlot but other tools are only detected in stackProvidingSlot so we check em both
+		if (!stackAtPlayerSlot.is(Items.AIR) && itemIgnored(stackAtPlayerSlot) || !stackProvidingSlot.is(Items.AIR) && itemIgnored(stackProvidingSlot))
+			return;
+
 		providingInv.extractItem(providingSlot, stackProvidingSlot.getCount(), false);
 		providingInv.insertItem(providingSlot, stackAtPlayerSlot, false);
 		
 		playerInv.setItem(playerSlot, stackProvidingSlot);
 	}
 
+	private boolean itemIgnored(ItemStack stack) {
```

Comment:
this should be computed on config change rather than every time

Reply:
it done